src/main/java/com/minecolonies/coremod/colony/buildings/BuildingTownHall.java

```diff
@@ -82,7 +82,7 @@ public void onUpgradeComplete(final int newLevel)
      */
     public void addPermissionEvent(final PermissionEvent event)
     {
-        if(getBuildingLevel() >= 1)
+        if(getBuildingLevel() >= 1 && !permissionEvents.contains(event))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)